lib/cartopy/crs.py

```diff
@@ -85,21 +594,54 @@ class Projection(CRS, metaclass=ABCMeta):
         'MultiPolygon': '_project_multipolygon',
     }
 
-    @abstractproperty
-    def boundary(self):
-        pass
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.bounds = None
+        if self.area_of_use:
+            # Convert lat/lon bounds to projected bounds.
+            # Geographic area of the entire dataset referenced to WGS 84
+            # NB. We can't use a polygon transform at this stage because
+            # that relies on the existence of the map boundary... the very
+            # thing we're trying to work out! ;-)
+            x0 = self.area_of_use.west
+            x1 = self.area_of_use.east
+            y0 = self.area_of_use.south
+            y1 = self.area_of_use.north
+            lons = np.array([x0, x0, x1, x1])
+            lats = np.array([y0, y1, y1, y0])
+            points = self.transform_points(self.geodetic_crs, lons, lats)
```

Comment:
This might be helpful here: https://pyproj4.github.io/pyproj/stable/api/transformer.html#pyproj.transformer.Transformer.transform_bounds